airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -75,15 +75,10 @@ def unify_bucket_name_and_key(func: T) -> T:
     def wrapper(*args, **kwargs) -> T:
         bound_args = function_signature.bind(*args, **kwargs)
 
-        def get_key_name() -> Optional[str]:
-            if 'wildcard_key' in bound_args.arguments:
-                return 'wildcard_key'
-            if 'key' in bound_args.arguments:
-                return 'key'
+        key_name = 'wildcard_key' if 'wildcard_key' in bound_args.arguments else 'key'
+        if key_name not in bound_args.arguments:
```

Comment:
seems like unnecessary check

Reply:
Oops. This is totally a typo! Fixing now.

Should have been `if 'bucket' not in ...`.

Reply:
How so? If you do not pass 'wildcard_key' nor 'key', this will be true, and raise exception.

Reply:
```python
        if 'wildcard_key' in bound_args.arguments:
            key_name = 'wildcard_key'
        elif 'key' in bound_args.arguments:
            key_name = 'key'
        else:
            raise ValueError('Missing key parameter!')
```

Would this be easier to follow?

Reply:
I agree with @uranusjr, I don't see a reason to make this code more cryptic in this particular case, just to save a few lines. I'd use the same cascading if statements that were present before your changes (and that @uranusjr is suggesting).

Reply:
Yes, let me do that.